keras/backend/torch/core.py

```diff
@@ -436,8 +436,63 @@ def unstack(x, num=None, axis=0):
     return x.unbind(axis)
 
 
-def custom_gradient(fun):
-    # TODO: Support this function
-    raise NotImplementedError(
-        "`custom_gradient` is not supported with torch backend"
-    )
+class custom_gradient:
```

Comment:
To keep the API consistent across backends, we may want to make this a functions `def custom_gradient(fun)` that instantiates `CustomGradientFunction` and calls it later when the function is called?

Reply:
@fchollet Yes. The `@custom_gradient` in `torch/core.py` acts as a decorator or `custom_gradient(fun)` as you said. It takes the function `fun` and returns a `torch.autograd.Function` instance (that is, a function whose gradient formula is now known to PyTorch framework). 

Additionally, `CustomGradientFunction` class has `forward()` and `backward()` functions with our required forward and backward definitions, as usually custom definitions are defined in PyTorch: https://pytorch.org/tutorials/beginner/examples_autograd/two_layer_net_custom_function.html

The effort was to unify the syntax and the method of creating new functions with custom gradients, across all backends. Since, PyTorch doesn't suppors decorator method of defining new gradients, I had to write a custom one for that.

For testing purposes, the second colab notebook uses my fork of Keras (because that has Custom Gradients implemented) and all frameworks one by one to test it. Just ensure to restart the Colab instance once to test another framework as backend after testing one.